### PR TITLE
Rename contrib to extensions

### LINKED ISSUE #1100: Rename `contrib` and `sdk_contrib`
`contrib` is causing a bit of confusion around the intention.  Perhaps something like `util` or `ext` instead?

### PR DESCRIPTION
**Pull request by iNikem**
Closes #1100

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,11 +17,19 @@ This project contains the following top level components:
...
   * [metrics](api/src/main/java/io/opentelemetry/metrics/): Metrics layer.
+* [extensions](extensions/): Additional extensions API, which are not part of the core API.
 * [sdk](sdk/): The reference implementation complying to the OpenTelemetry API.
+* [sdk_extensions](sdk_extensions/): Additional extensions to SDK.
 * [OpenTracing shim](opentracing_shim/): A bridge layer from OpenTelemetry to the OpenTracing API.
 
 We would love to hear from the larger community: please provide feedback proactively.
 
+### Note about extensions
+Both API and SDK extensions consist of various component which, for various reasons, are not deemed
```
Comment (arminru) [inline README.md:28]:
```suggestion
### Note about extensions

Both API and SDK extensions consist of various components which, for various reasons, are not deemed
```

Comment (arminru) [inline README.md:28]: A reader might wonder what these "various reasons" are. Could you list them there?

**Comment by carlosalberto** [comment]
Overall good, but I'm wondering whether we should adjust the names for some of these packages, i.e. `OpenTelemetry Contrib Auto Annotations` became `OpenTelemetry Extension Auto Annotations` doesn't quite *feel* right ;)

**Comment by iNikem** [comment]
> Overall good, but I'm wondering whether we should adjust the names for some of these packages, i.e. `OpenTelemetry Contrib Auto Annotations` became `OpenTelemetry Extension Auto Annotations` doesn't quite _feel_ right ;)

All suggestions are welcomed.

File: extensions/auto_annotations/build.gradle [hunk 2]
```diff
@@ -5,8 +5,8 @@ plugins {
     id "ru.vyarus.animalsniffer"
 }
 
-description = 'OpenTelemetry Contrib Auto Annotations'
-ext.moduleName = "io.opentelemetry.contrib.auto.annotations"
+description = 'OpenTelemetry Extension Auto Annotations'
```
Comment (jkwatson) [inline extensions/auto_annotations/build.gradle:8]:
```suggestion
description = 'OpenTelemetry Extensions for Auto Annotations'
```

File: extensions/trace_propagators/build.gradle [hunk 3]
```diff
@@ -6,8 +6,8 @@ plugins {
     id "me.champeau.gradle.jmh"
 }
 
-description = 'OpenTelemetry Contrib Trace Propagators'
-ext.moduleName = "io.opentelemetry.contrib.trace.propagation"
+description = 'OpenTelemetry Extension Trace Propagators'
```
Comment (jkwatson) [inline extensions/trace_propagators/build.gradle:9]:
```suggestion
description = 'OpenTelemetry Extension : Trace Propagators'
```

File: sdk_extensions/async_processor/build.gradle [hunk 4]
```diff
@@ -5,8 +5,8 @@ plugins {
     id "ru.vyarus.animalsniffer"
 }
 
-description = 'OpenTelemetry SDK Contrib'
-ext.moduleName = "io.opentelemetry.sdk.contrib.trace.export"
+description = 'OpenTelemetry SDK Extension'
```
Comment (jkwatson) [inline sdk_extensions/async_processor/build.gradle:8]:
```suggestion
description = 'OpenTelemetry SDK Extension: Async SpanProcessor'
```

**Comment by jkwatson** [comment]
looks like it needs a rebase. I like it, though!

**Comment by iNikem** [comment]
> looks like it needs a rebase. I like it, though!

Done.

**Comment by carlosalberto** [comment]
Since it's a relatively important change, lets wait a little while before merging it, so people can comment/review ;)

**Comment by iNikem** [comment]
Ready to merge?

**Pull request merged**